infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -219,11 +236,22 @@ public void sendWithBuffering(List<LogReplicationEntryMsg> dataToSend, String me
     /**
      * Resend the messages in the queue if they have timed out.
      */
-    public LogReplicationEntryMsg resend() {
+    public LogReplicationEntryMsg resend(boolean isLogEntry) {
         LogReplicationEntryMsg ack = null;
         boolean force = false;
+
         try {
             ack = processAcks();
+
+            if (isLogEntry && isBackpressureActive) {
```

Comment:
Are we going to come to this block incase we don't receive an ACK?
processAcks() waits for sometime and then times out..throwing a timeout exception in the process. So we won't be coming here right?

Reply:
Might have to move it to the for block below

Reply:
sleeping for a smaller amount between each resend in the loop is probably better retry strategy, will take that up